### PR TITLE
Always install ebpfsvc

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Part of #4427

## Description

The ebpfsvc will host the logic to validate the signature on native BPF programs and as such is now a required component.

## Testing

CI/CD

## Documentation

No.

## Installation

Yes, ebpfsvc is no longer optional.

### CODE REVIEW

**Review by dthaler — DISMISSED** [review]

File: .azure/OneBranch.PullRequest.yml [hunk 1]
```diff
@@ -59,7 +59,6 @@ jobs:
           generate_release_package: true
           build_msi: true
           build_nuget: true
-          build_options: /p:ReleaseJIT=True
```
Comment (saxena-anurag) [inline .azure/OneBranch.PullRequest.yml:62]: just to confirm: for a regular `Release` build, interpreter will not be included?

File: ebpfsvc/rpc_api.cpp [hunk 2]
```diff
@@ -41,4 +42,12 @@ ebpf_server_verify_and_load_program(
 
     ebpf_clear_thread_local_storage();
     return result;
+#else
+    // JIT and interpreter are disabled.
```
Comment (saxena-anurag) [inline ebpfsvc/rpc_api.cpp:46]: not sure if we already have negative tests for this code path, but if not present, should we add those? (to test the negative case)

File: .azure/OneBranch.PullRequest.yml (hunk 1 above)
Comment (Alan-Jowett) [reply .azure/OneBranch.PullRequest.yml:62]: Correct

File: ebpfsvc/rpc_api.cpp (hunk 2 above)
Comment (Alan-Jowett) [reply ebpfsvc/rpc_api.cpp:46]: Planning on removing the RPC method so this is unreachable.

File: scripts/check_msi_installation.ps1 [hunk 3]
```diff
@@ -14,7 +14,7 @@ $InstallPath = "$env:ProgramFiles\ebpf-for-windows";
 # - The expected file lists for each build artifact
 $buildArtifactParams = @{
     "Build-x64_Debug" = @{
-        "InstallComponents" = "ADDLOCAL=eBPF_Runtime_Components,eBPF_Runtime_Components_JIT"
+        "InstallComponents" = "ADDLOCAL=eBPF_Runtime_Components"
```
Comment (matthewige) [inline scripts/check_msi_installation.ps1:17]: github is not letting me put this comment below, but I think there is one more part below that needs updating - we check for eBPF_Runtime_Components_JIT to determine if we need to check ebpfsvc

**Review by matthewige — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]
